File: lab/python/arrays_e_listas/maior_e_menor_array.py
def maior_e_menor(*valores):
    print(f"Maior = {max(valores)}")
    print(f"Menor = {min(valores)}")


def maior_e_menor2(valores):
    if not isinstance(valores, list):
        print("Tipo de dado inválido!")
    elif not valores:
        print("Vazio!")
    else:
        print(f"Maior = {max(valores)}")
        print(f"Menor = {min(valores)}")


def maior_e_menor3(valores):
    if not isinstance(valores, list):
        print("Tipo de dado inválido!")
    elif not valores:
        print("Vazio!")
    else:
        maior = valores[0]
        menor = valores[0]
        for i in range(len(valores)):
            if valores[i] > maior:
                maior = valores[i]
            elif valores[i] < menor:
                menor = valores[i]
        print(f"Maior = {maior}")
        print(f"Menor = {menor}")


def maior_e_menor4(valores):
    if not isinstance(valores, list):
        print("Tipo de dado inválido!")
    elif not valores:
        print("Vazio!")
    else:
        ordenada = sorted(valores)
        menor = ordenada[0]
        maior = ordenada[-1]
        print(f"Maior = {maior}")
        print(f"Menor = {menor}")


maior_e_menor(3, 8, 1, 9, 2)
print()
maior_e_menor2([3, 8, 1, 9, 2])
print()
maior_e_menor3([3, 8, 1, 9, 2])
print()
maior_e_menor4([3, 8, 1, 9, 2])

File: lab/javascript/arrays_e_listas/maior_e_menor_array.js
function maior_e_menor(...valores) {
  console.log(`Maior = ${Math.max(...valores)}`);
  console.log(`Menor = ${Math.min(...valores)}`);
}

function maior_e_menor2(valores) {
  if (!Array.isArray(valores)) {
    console.log("Tipo de dado inválido!");
  } else if (valores.length === 0) {
    console.log("Vazio!");
  } else {
    console.log(`Maior = ${Math.max(...valores)}`);
    console.log(`Menor = ${Math.min(...valores)}`);
  }
}

function maior_e_menor3(valores) {
  if (!Array.isArray(valores)) {
    console.log("Tipo de dado inválido!");
  } else if (valores.length === 0) {
    console.log("Vazio!");
  } else {
    let maior = valores[0];
    let menor = valores[0];
    for (let i = 1; i < valores.length; i++) {
      if (valores[i] > maior) maior = valores[i];
      else if (valores[i] < menor) menor = valores[i];
    }
    console.log(`Maior = ${maior}`);
    console.log(`Menor = ${menor}`);
  }
}

function maior_e_menor4(valores) {
  if (!Array.isArray(valores)) {
    console.log("Tipo de dado inválido!");
  } else if (valores.length === 0) {
    console.log("Vazio!");
  } else {
    const ordenada = [...valores].sort((a, b) => a - b);
    const menor = ordenada[0];
    const maior = ordenada[ordenada.length - 1];
    console.log(`Maior = ${maior}`);
    console.log(`Menor = ${menor}`);
  }
}

// -------------------- Testes --------------------
maior_e_menor(3, 8, 1, 9, 2);
console.log();
maior_e_menor2([3, 8, 1, 9, 2]);
console.log();
maior_e_menor3([3, 8, 1, 9, 2]);
console.log();
maior_e_menor4([3, 8, 1, 9, 2]);
